source/Missions/Services/Agents/Messages/AgentMovementDelta.cs

```diff
@@ -0,0 +1,165 @@
+﻿using Missions.Services.Agents.Packets;
+using ProtoBuf;
+using System;
+using TaleWorlds.Library;
+using TaleWorlds.MountAndBlade;
+
+namespace Missions.Services.Agents.Messages
+{
+    /// <summary>
+    /// Represents the delta of an <see cref="Agent"/>'s movement between the time it was created and the time it was last updated.
+    /// </summary>
+    [ProtoContract(SkipConstructor = true)]
+    public class AgentMovementDelta
```

Comment:
This class is meant to replace the `MovementPacket` class. If there is some reason this isn't easily completable let me know.

Make this class inherit from IPacket